test/integration/backward_compatible/parallel/serialization/compact/CompactUtil.js

```diff
@@ -1486,6 +1487,8 @@ if (TestUtil.isClientVersionAtLeast('5.1.0')) {
         } catch (e) {
             if (e instanceof SchemaNotReplicatedError) {
                 await schemaService.put(e.schema);
+            } else if (e instanceof HazelcastSerializationError) {
+                throw new HazelcastSerializationError(e.message, e.cause, e.serverStackTrace);
```

Comment:
This code is not needed. any reasoning? You are catching and rethrowing the same exception.

Reply:
When we don't catch this error it continuously executes the same function (see line [1493](https://github.com/hazelcast/hazelcast-nodejs-client/pull/1365/files#diff-d952c81eb5a44c38d522d83728d20c98c52e1bf57f9c2af12793dcbfafffa717R1493)) until out of memory. We are using this function only for testing purposes.

Reply:
Ok makes sense. Then we can change the logic a little bit to be as follows.

```js
try {
      return serializationService.toData(obj);
} catch (e) {
    if (e instanceof SchemaNotReplicatedError) {
        await schemaService.put(e.schema);  
        return serialize(serializationService, schemaService, obj);
    }
    throw e;
}
```

Note that `await` is not needed in while returning recursively. It was unnecessary.

Reply:
Refactored at https://github.com/hazelcast/hazelcast-nodejs-client/pull/1365/commits/1d60244de304441b291ab50d5177b263bfc11107.